### PR TITLE
custom fields: old data may be None

### PR DESCRIPTION
**Pull request by bpedersen2**
Happens on migrated events from 1.2.

### CODE REVIEW

**Comment by bpedersen2** [comment]
Stacktrace:
2017-09-14 15:11:21,426  ERROR    6d2dce201a91479e  indico.requestHandler     Request finished: AttributeError ('NoneType' object has no attribute 'get')
Traceback (most recent call last):
  File "/usr/home/indicoadm/indicov2/.indicov2/lib/python2.7/site-packages/indico/legacy/webinterface/rh/base.py", line 493, in process
    profile_name, res = self._do_process(profile)
  File "/usr/home/indicoadm/indicov2/.indicov2/lib/python2.7/site-packages/indico/legacy/webinterface/rh/base.py", line 464, in _do_process
    res = self._process()
  File "/usr/home/indicoadm/indicov2/.indicov2/lib/python2.7/site-packages/indico/modules/events/registration/controllers/__init__.py", line 47, in _process
    if form.validate_on_submit():
  File "/usr/home/indicoadm/indicov2/.indicov2/lib/python2.7/site-packages/flask_wtf/form.py", line 161, in validate_on_submit
    return self.is_submitted() and self.validate()
  File "/usr/home/indicoadm/indicov2/.indicov2/lib/python2.7/site-packages/indico/web/forms/base.py", line 136, in validate
    valid = super(IndicoForm, self).validate()
  File "/usr/home/indicoadm/indicov2/.indicov2/lib/python2.7/site-packages/wtforms/form.py", line 310, in validate
    return super(Form, self).validate(extra)
  File "/usr/home/indicoadm/indicov2/.indicov2/lib/python2.7/site-packages/wtforms/form.py", line 152, in validate
    if not field.validate(self, extra):
  File "/usr/home/indicoadm/indicov2/.indicov2/lib/python2.7/site-packages/wtforms/fields/core.py", line 204, in validate
    stop_validation = self._run_validation_chain(form, chain)
  File "/usr/home/indicoadm/indicov2/.indicov2/lib/python2.7/site-packages/wtforms/fields/core.py", line 224, in _run_validation_chain
    validator(form, self)
  File "/usr/home/indicoadm/indicov2/.indicov2/lib/python2.7/site-packages/indico/modules/events/registration/fields/choices.py", line 113, in _check_number_of_places
    .data_by_field[self.form_item.id].data.get(k, 0))
AttributeError: 'NoneType' object has no attribute 'get'

**Comment by ThiefMaster** [comment]
This sounds more like something we should fix in the migration script. I think"no choice" is stored as `{}` so we should do the same when migrating data.

This is probably just a missing `rv['data'] = {}` between these two lines of the single-choice field data migration:

        elif not selected:
            return

**Comment by bpedersen2** [comment]
Well, fixing the migration is probably a good idea. But will that help for an alreday migrated site?

**Comment by ThiefMaster** [comment]
Won't be too hard to write a SQL query that updates the bad data

File: indico/modules/events/registration/fields/choices.py [hunk 1]
```diff
@@ -108,7 +108,7 @@ def _check_number_of_places(form, field):
                     places_limit = choice.get('places_limit')
                     places_used_dict = self.get_places_used()
                     places_used_dict.setdefault(k, 0)
-                    if form.modified_registration:
+                    if form.modified_registration and old_data.data:
                         places_used_dict[k] -= (form.modified_registration
                                                     .data_by_field[self.form_item.id].data.get(k, 0))
```
Comment (bpedersen2) [inline indico/modules/events/registration/fields/choices.py]: Question: Is written in this form on purpose to refetch the data jus there, or can this be written as  old_data.data.get...

Comment (ThiefMaster) [reply indico/modules/events/registration/fields/choices.py]: I was asking myself the same question when looking at this code ;)

**Comment by ThiefMaster** [comment]
Here's an SQL query you can run via `psql` or any other postgres client to fix the data:

```sql
UPDATE event_registration.registration_data rd
SET data = '{}'::json
FROM event_registration.form_field_data ffd, event_registration.form_items fi
WHERE ffd.id = rd.field_data_id AND
      fi.id = ffd.field_id AND
      fi.input_type = 'single_choice' AND
      rd.data::jsonb = 'null'::jsonb;
```

**Comment by bpedersen2** [comment]
Thanks ;)

**Pull request closed without merging**